src/lib/MetadataAPI.js

```diff
@@ -0,0 +1,56 @@
+import api from './api';
+import { getHeaders } from './util';
+
+// Map of extensions to global metadata template
+const EXTENSIONS_TO_TEMPLATES = {
+    dwg: 'autocad'
+};
+
+const MetadataAPI = {
+    /**
+     * Gets the global metadata template for the specified file
+     * @param {string} file.id - The file id
+     * @param {string} file.extension - The file extension
+     * @param {Object} options - options object
+     * @return {Promise} Promise is resolved or rejected based on response
+     */
+    get({ id, extension }, options = {}) {
+        const fileType = EXTENSIONS_TO_TEMPLATES[extension];
+
+        if (!fileType) {
+            return Promise.reject(new Error(`${extension} is not a supported extension`));
+        }
+
+        const { apiHost, token } = options;
+
+        return new Promise((resolve, reject) => {
```

Comment:
Looks like we can just `return api.get` rather than wrapping in a new promise. You can re-throw the error in `.catch` if you want the parent promise to hear it as well.

Reply:
I wanted to abstract away the logic of http status 404 being ok so that calling code doesn't need to worry about it. Is there another way to resolve the promise in the catch logic after detecting the 404 situation?